workspace.go

```diff
@@ -111,10 +123,15 @@ type Workspace struct {
 	RunsCount            int                   `jsonapi:"attr,workspace-kpis-runs-count"`
 
 	// Relations
-	AgentPool    *AgentPool    `jsonapi:"relation,agent-pool"`
-	CurrentRun   *Run          `jsonapi:"relation,current-run"`
-	Organization *Organization `jsonapi:"relation,organization"`
-	SSHKey       *SSHKey       `jsonapi:"relation,ssh-key"`
+	AgentPool                  *AgentPool                `jsonapi:"relation,agent-pool"`
+	CurrentRun                 *Run                      `jsonapi:"relation,current-run"`
+	Organization               *Organization             `jsonapi:"relation,organization"`
+	RemoteStateConsumerDetails *RemoteStateConsumerLinks `jsonapi:"relation,remote-state-consumers"`
```

Comment:
Hmmm I don't think this needs to be unmarshaled with a response at all; it looks like `jsonapi` provides a `RelationshipLinkable` interface such that I _think_ you can just define a function to build the right URL, like this:

```go
func (w Workspace) JSONAPIRelationshipLinks(relation string) *Links {
	if relation == "remote-state-consumers" {
		return &Links{
			"related": fmt.Sprintf("<the right url>"),
		}
	}
	return nil
}
```

https://github.com/google/jsonapi#links

? Unless I'm misunderstanding

Reply:
Implimenting the `Linkable` interface only works for [Marshalling](https://github.com/hashicorp/jsonapi/blob/master/response.go#L468-L478). Which makes sense in that you want to customize what links get sent during a request.

But for unmarshalling, it reads the `links` annotation (whether on a relation or directly on the struct), and [unmarshalls](https://github.com/hashicorp/jsonapi/blob/master/request.go#L311) it [into](https://github.com/hashicorp/jsonapi/blob/master/request.go#L469) an arbitrary [map[string]interface](https://github.com/hashicorp/jsonapi/blob/master/request.go#L541)

Reply:
Ahhh I see the disconnect here.

I'm looking at it like the actual raw relationships link doesn't need to be fetchable, unmarshalled data. With the API you've built to interact with those remote state consumers, I wonder if we necessarily need to include every bit of link metadata at all, especially if the client itself ignores it (at marshalling, like you said).

There _are_ cases where we probably will need to actually deserialize the raw link and utilize the awesome work you did in the jsonapi package fork, like [the blob storage upload link for policy set versions](https://www.terraform.io/docs/cloud/api/policy-sets.html#create-a-policy-set-version), where someone may want to take the raw url and feed it to some external pipeline to upload to. 

For here, it's just a reference to domain objects still within the domain of the client, so the link is fairly moot. 

I should add that my biggest hesitation here is stamping `XDetails` as the convention for exposing any relationships with links in the future; something feels like it should be generalized more in to some metadata abstraction. At the very least, I think we should remove this small bit (just because it's not really necessary) and continue discussion to add links in general, maybe add it later?

Reply:
All those points make sense, will remove and if we see a need later on, we can add it back and have more context as to why we'd use it.